### PR TITLE
Disallow no-op calls for grantPermission, revokePermission, reno…

### LINKED ISSUE #2379: Revert on grant/revoke/renounce no-op
We currently allow granting a permission that is already held, as well as revoking and renouncing one that is not held. This means user error will go undetected - we should instead revert and fail loudly.

### PR DESCRIPTION
**Pull request by ylv-io**
# Description

This pull request modifies the following functions of the TimelockAuthroizer:
* grantPermission
* revokePermission
* renouncePermission

Previously, these functions did not revert even if they did not modify any permission or make any changes. However, we have decided that no one should attempt to revoke or renounce a permission they do not have or grant permission to an account that already has it. Therefore, we now revert in these cases.

## Type of change

- [ ] Bug fix 
- [ ] New feature 
- [x] Breaking change 
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [x] The diff is legible and has no extraneous changes
- [x] Complex code has been commented, including external interfaces
- [x] Tests are included for all code paths
- [x] The base branch is either `master`, or there's a description of how to merge

## Issue Resolution

Closes #2379

### CODE REVIEW

File: pkg/vault/test/authorizer/TimelockAuthorizerPermissions.test.ts [hunk 1]
```diff
@@ -894,6 +816,7 @@ describe('TimelockAuthorizer permissions', () => {
       });
 
       it('does not revoke any other permissions when executed', async () => {
+        await authorizer.grantPermission(ACTION_1, user, WHERE_1, { from: root });
```
Comment (nventuro) [inline pkg/vault/test/authorizer/TimelockAuthorizerPermissions.test.ts:819]: Same here

**Review by nventuro — COMMENTED** [review]
Looks great other than a small comment in one of the tests!

File: pkg/vault/test/authorizer/TimelockAuthorizerPermissions.test.ts [hunk 2]
```diff
@@ -872,6 +793,7 @@ describe('TimelockAuthorizer permissions', () => {
       });
 
       it('can be executed after the expected delay', async () => {
+        await authorizer.grantPermission(ACTION_1, user, WHERE_1, { from: root });
```
[the remark this answers is not in the document]
Comment (ylv-io) [reply pkg/vault/test/authorizer/TimelockAuthorizerPermissions.test.ts:796]: It is needed. It is impossible to revoke a permission if it was not granted before.

[the remark this answers is not in the document]
Comment (nventuro) [reply pkg/vault/test/authorizer/TimelockAuthorizerPermissions.test.ts:796]: Ahhh that makes sense! I'll add a small comment

Comment (nventuro) [inline pkg/vault/test/authorizer/TimelockAuthorizerPermissions.test.ts:796]:
```suggestion
        // Grant the permission so we can later revoke it
        await authorizer.grantPermission(ACTION_1, user, WHERE_1, { from: root });
```

File: pkg/vault/test/authorizer/TimelockAuthorizerPermissions.test.ts (hunk 1 above)
Comment (nventuro) [inline pkg/vault/test/authorizer/TimelockAuthorizerPermissions.test.ts:819]:
```suggestion
        // Grant the permission so we can later revoke it
        await authorizer.grantPermission(ACTION_1, user, WHERE_1, { from: root });
```

**Review by nventuro — APPROVED** [review]

**Review by alokeutpal — APPROVED** [review]

**Comment by alokeutpal** [comment]
yes I don't understand the whole thing, if I could understand a little if I got Bengali language

**Pull request merged**